### PR TITLE
Update deployment patch route status code

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR updates deployment patch route status code. When an async task is created, 202 is returned.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/routes/deployment/api.py [hunk 1]
```diff
@@ -72,13 +72,16 @@ async def get_deployment(request: Request, deployment_id: PydanticObjectId) -> D
...
 ) -> Optional[Task]:
     """
     Update Deployment
     """
     controller = request.state.app_container.deployment_controller
     task: Task = await controller.update_deployment(document_id=deployment_id, data=data)
+    if isinstance(task, Task):
+        # if task is returned, it means the deployment is being updated asynchronously
+        response.status_code = HTTPStatus.ACCEPTED
     return task
```
Comment (gee-senbong) [inline featurebyte/routes/deployment/api.py:85]: Update this part so that it is aligned with the main branch to return 202 when an async job is created.

File: .github/workflows/test_gen.yaml [hunk 2]
```diff
@@ -194,8 +194,11 @@ jobs:
       matrix:
         python-version:
         - 3.8.12
+        spark-version:
+        - 3.3.1
+        - 3.5.0
```
Comment (YS-L) [inline .github/workflows/test_gen.yaml:199]: Curious if there is a reason behind the workflows changes? They seem unrelated to the main change in the PR.

File: featurebyte/routes/deployment/api.py [hunk 3]
```diff
@@ -72,13 +72,16 @@ async def get_deployment(request: Request, deployment_id: PydanticObjectId) -> D
...
 async def update_deployment(
-    request: Request, deployment_id: PydanticObjectId, data: DeploymentUpdate
+    request: Request, deployment_id: PydanticObjectId, data: DeploymentUpdate, response: Response
 ) -> Optional[Task]:
     """
     Update Deployment
     """
     controller = request.state.app_container.deployment_controller
     task: Task = await controller.update_deployment(document_id=deployment_id, data=data)
+    if isinstance(task, Task):
```
Comment (YS-L) [inline featurebyte/routes/deployment/api.py:82]: Is the `task: Task` type annotation above accurate? It seems to return either a Task or something else, otherwise it seems like the condition here would always evaluate to True.

**Review by YS-L — APPROVED** [review]

File: .github/workflows/publish-dev_gen.yaml [hunk 4]
```diff
@@ -91,36 +93,13 @@ jobs:
...
-#          ${{ env.SERVER_IMAGE_REPO }}:${{ env.SHORT_HASH }}
-#          ${{ env.SERVER_IMAGE_REPO }}:${{ env.DEV_VERSION }}
-#          ${{ env.SERVER_IMAGE_REPO }}:latest
-#        context: .
-#        platforms: |-
-#          linux/amd64
-#          linux/arm64
-#        file: docker/Dockerfile
-#        cache-from: type=registry,ref=${{ env.SERVER_IMAGE_REPO }}:latest
-#        cache-to: type=inline
```
Comment (bushwhackr) [inline .github/workflows/publish-dev_gen.yaml:116]: This one doesnt matter. It will only run on pushes to main branch.
Essentially this workflow is a NOP when merged to release branch

File: .github/workflows/daily-cron_gen.yaml [hunk 5]
```diff
@@ -59,13 +59,15 @@ jobs:
     - name: Configure Poetry
       run: poetry config virtualenvs.in-project true
     - name: Install packages needed for build
-      run: sudo apt-get install libkrb5-dev libsasl2-dev libpython3-dev g++ gcc
+      run: |-
+        sudo apt-get update
```
Comment (bushwhackr) [inline .github/workflows/daily-cron_gen.yaml:63]: This line is the only relevant one to get it running. Essentially, gh runners just need their apt cache refreshed.

**Review by bushwhackr — APPROVED** [review]

File: .github/workflows/test_gen.yaml (hunk 2 above)
Comment (gee-senbong) [reply .github/workflows/test_gen.yaml:199]: Need to update the workflow here so that the tests won't fail due to linux image not get updated. I checked out the workflow from main branch to this release branch.
